internal/common_context.go

```diff
@@ -151,7 +157,13 @@ type QueryParamInvalidTypeError struct {
 }
 
 func (e QueryParamInvalidTypeError) Error() string {
-	return fmt.Errorf("param %s=%s is not of type %s: %w", e.ParamName, e.ParamValue, e.ExpectedType, e.Err).Error()
+	return fmt.Errorf("query param %s=%s is not of type %s: %w", e.ParamName, e.ParamValue, e.ExpectedType, e.Err).Error()
```

Comment:
My comment is more about the code than the idea of your PR.

Also, I feel like you may have copy pasted existing code.

But I would have done this


```suggestion
	return fmt.Sprintf("query param %s=%s is not of type %s: %v", e.ParamName, e.ParamValue, e.ExpectedType, e.Err)
```

Reply:
Ope yep I just copy pasta’s everywhere